### ISSUE TITLE
localAddress option not settable

### ISSUE THREAD

**Issue by thomaschaaf** [forwardemail/superagent#1280]
I would like to use superagent and have the requirement that I would like to use a certain network interface for my requests. Currently as far as I can tell there is no way to provide superagent with a `localAddress` option.

Is this something that is not built in on purpose or would a PR for this be appreciated?

**Comment by pornel** [comment]
Superagent doesn't have this functionality. Mainly because it's not a common need, and it doesn't have equivalent on the browser side (we try to maintain parity).

I'm not sure what to do about this. The use-case is valid, but probably rare.

On browser side we expose the XHR object. Maybe we can just expose the Node's request object too? (as a "use on your own risk" API for such rare cases). @focusaurus WDYT?

**Comment by focusaurus** [comment]
I agree it's a rare but valid use case so a tough call. If we expose the
request object can you configure localAddress on it after it's already been
constructed?

**Comment by thomaschaaf** [comment]
It would need to be set somewhere around here https://github.com/visionmedia/superagent/blob/d55adf1a86623247ee19a8bb75024f5098ed42a2/lib/node/index.js#L595

The `localAddress` has to be set when initializing the request according to: https://nodejs.org/api/http.html#http_http_request_options_callback

So sadly it can not be set afterwards.

**Comment by titanism** [comment]
localAddress must be supplied at http(s).request construction and is not exposed today. Add a Node-only documented option/setter with coverage; browsers have no equivalent.

v10.4.0 is available on npm: https://www.npmjs.com/package/superagent/v/10.4.0
GitHub release: https://github.com/forwardemail/superagent/releases/tag/v10.4.0

Consider supporting our efforts to maintain this project by using Forward Email at https://forwardemail.net.  We run the only 100% open source and 3rd party audited email service in the world.

**Comment by titanism** [comment]
Closing this for now. If this still affects v10.4.0, please open a new issue with a current reproduction.

v10.4.0 is available on npm: https://www.npmjs.com/package/superagent/v/10.4.0
GitHub release: https://github.com/forwardemail/superagent/releases/tag/v10.4.0

Consider supporting our efforts to maintain this project by using Forward Email at https://forwardemail.net.  We run the only 100% open source and 3rd party audited email service in the world.
